### ISSUE TITLE
Course and Section metadata missing from question card

### ISSUE THREAD

**Issue by bree-z** [openlab-at-city-tech/webworkqa#208]
Only the question name appears at the bottom of the question card, but not the course or section - see screenshot below. 

The course and section metadata appears for some older questions (May 2020), but not for questions posted in October 2020 or currently.

![MissingCourseSection](https://user-images.githubusercontent.com/7189824/131500354-5babbf97-285d-4ae5-8c51-7ff97ba67c44.png)

**Comment by boonebgorges** [comment]
Is this on the production site or the test site? This particular piece of metadata has been juggled a number of times with the Rederly changes.

**Comment by bree-z** [comment]
This is on https://hardg.com/webwork/.

**Comment by boonebgorges** [comment]
See #195 

@drdrew42 Here's what I'm seeing coming from http://146.111.135.122/webwork2/WW-Dev/Polynomials_-_Division/1/

```
[webwork_user] => studentB
    [problem_set] => Polynomials_-_Division
    [problem_number] => 1
    [problem_id] => 
    [problem_text] => 
    [course] => 
    [emailableURL] => http://146.111.135.122/webwork2/WW-Dev/Polynomials_-_Division/1/?effectiveUser=studentB
    [randomSeed] => 1542
    [notifyAddresses] => =?UTF-8?B?QWRtaW5pc3RyYXRvcg==?= ;=?UTF-8?B?UHJvZi4gSW5zdHJ1Y3Rvcg==?= ;=?UTF-8?B?VG9kZCBTdGFtYmF1Z2g=?= ;=?UTF-8?B?Qm9vbmUgR29yZ2Vz?=
    [studentName] => Student B
```

`course` is empty. webworkqa is actually looking for `courseId`, but this is not present at all. According to https://github.com/openwebwork/webwork2/pull/1212, `courseId` is the canonical source. Could you make sure that the dev site is running the correct version of WW? Or if I'm looking at the wrong WW dev site, can someone tell me where the right one is?

**Comment by drdrew42** [comment]
It might be the case that the dev server needs to be pulled up to the latest release… I’ll check at 6pm when my class ends – thanks for the heads up!

From: Boone Gorges ***@***.***>
Reply-To: openlab-at-city-tech/webworkqa ***@***.***>
Date: Tuesday, August 31, 2021 at 1:03 PM
To: openlab-at-city-tech/webworkqa ***@***.***>
Cc: Kenneth Parker ***@***.***>, Mention ***@***.***>
Subject: Re: [openlab-at-city-tech/webworkqa] Course and Section metadata missing from question card (#208)


See #195<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_openlab-2Dat-2Dcity-2Dtech_webworkqa_issues_195&d=DwMCAg&c=pRW6ZPn_LDv0DnDIAK65Ad0CA4hBS-2mAmNa2_oHfF0&r=fkJzUaFivH5t4sLPFjBaqIXmPHggMcfrb3tOho8Rbqs&m=XfQcurCHHZtU_aAFRLCSDKiukPHjoN6IsWwAGXbQAEE&s=FNVkuz38IABjK1BQ6L1i74iJDcSM7P6nfmys8koJgys&e=>

@drdrew42<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_drdrew42&d=DwMCAg&c=pRW6ZPn_LDv0DnDIAK65Ad0CA4hBS-2mAmNa2_oHfF0&r=fkJzUaFivH5t4sLPFjBaqIXmPHggMcfrb3tOho8Rbqs&m=XfQcurCHHZtU_aAFRLCSDKiukPHjoN6IsWwAGXbQAEE&s=HAhIvv8G4oCU72nNwHcNWmw1Gi6Xe0xjF9Q7h4W9RFw&e=> Here's what I'm seeing coming from http://146.111.135.122/webwork2/WW-Dev/Polynomials_-_Division/1/<https://urldefense.proofpoint.com/v2/url?u=http-3A__146.111.135.122_webwork2_WW-2DDev_Polynomials-5F-2D-5FDivision_1_&d=DwMCAg&c=pRW6ZPn_LDv0DnDIAK65Ad0CA4hBS-2mAmNa2_oHfF0&r=fkJzUaFivH5t4sLPFjBaqIXmPHggMcfrb3tOho8Rbqs&m=XfQcurCHHZtU_aAFRLCSDKiukPHjoN6IsWwAGXbQAEE&s=swqvH46I4RmhNa5_68ZCf4-jlAayV6rUXwcp8SQQVfs&e=>

[webwork_user] => studentB

    [problem_set] => Polynomials_-_Division

    [problem_number] => 1

    [problem_id] =>

    [problem_text] =>

    [course] =>

    [emailableURL] => http://146.111.135.122/webwork2/WW-Dev/Polynomials_-_Division/1/?effectiveUser=studentB<https://urldefense.proofpoint.com/v2/url?u=http-3A__146.111.135.122_webwork2_WW-2DDev_Polynomials-5F-2D-5FDivision_1_-3FeffectiveUser-3DstudentB&d=DwQCAg&c=pRW6ZPn_LDv0DnDIAK65Ad0CA4hBS-2mAmNa2_oHfF0&r=fkJzUaFivH5t4sLPFjBaqIXmPHggMcfrb3tOho8Rbqs&m=XfQcurCHHZtU_aAFRLCSDKiukPHjoN6IsWwAGXbQAEE&s=ulWINfpjE4_9N97m9R1_9LJ078uosYdashCne25wYgw&e=>

    [randomSeed] => 1542

    [notifyAddresses] => =?UTF-8?B?QWRtaW5pc3RyYXRvcg==?= ;=?UTF-8?B?UHJvZi4gSW5zdHJ1Y3Rvcg==?= ;=?UTF-8?B?VG9kZCBTdGFtYmF1Z2g=?= ;=?UTF-8?B?Qm9vbmUgR29yZ2Vz?=

    [studentName] => Student B

course is empty. webworkqa is actually looking for courseId, but this is not present at all. According to openwebwork/webwork2#1212<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_openwebwork_webwork2_pull_1212&d=DwMCAg&c=pRW6ZPn_LDv0DnDIAK65Ad0CA4hBS-2mAmNa2_oHfF0&r=fkJzUaFivH5t4sLPFjBaqIXmPHggMcfrb3tOho8Rbqs&m=XfQcurCHHZtU_aAFRLCSDKiukPHjoN6IsWwAGXbQAEE&s=xYjNc4ZFcpYYZgjkJTfgRWYQVvyk3nVKOJA4oWvu7Rw&e=>, courseId is the canonical source. Could you make sure that the dev site is running the correct version of WW? Or if I'm looking at the wrong WW dev site, can someone tell me where the right one is?

**Comment by drdrew42** [comment]
Okay, it turns out that somehow I botched that pull request and updated the wrong subroutine. (There is an ongoing effort to de-duplicate the WeBWorK code base -- but it's everywhere...) The actual change needs to be made elsewhere, which I've done for the dev server. I'll put in a new PR as a hotfix and let the dev team know this is relevant to the public release of the Q&A plugin.

**Comment by bree-z** [comment]
This looks fine now -- I see both the course and the problem set. 

@boonebgorges I just have one question about the filters - I only see two dropdowns, for course and problem set, but I assume this is just because there are no sections. If that sounds correct I think we can close this.

Thanks!

**Comment by boonebgorges** [comment]
That's correct - there's no concept of Sections anymore, after #195. Thanks for testing!
